### DISCUSSION TITLE
Visualization in MOOSE [DoFs]

### DISCUSSION THREAD

**Discussion by jinca** [idaholab/moose#19433] [Q&A Meshing] [answered]
Hello all, 
May I ask how many DoFs the MOOSE application and peacock can handle?
Does it have a maximum number? what about more than 121 millions of DoFs?
Thanks in advance!
Julita

**Comment by ykvishal** [comment, marked as answer]
It is better to limit peacock for small  simulations only. You can always run simulation via terminal and visualize output with [Paraview](https://www.paraview.org/). I ran few simulations with 3.5e9 DOFs, but you will need huge computational resources to run such large simulations.

**Comment by jinca** [comment]
Thank you for the answer. Could you please let me know if you used a supercomputer? My DoF is 10 to 8 instead of your test of 10 to 9.
Appreciate again your help, Julita

**Reply by ykvishal** [reply]
Yes, I always use HPC for running MOOSE  simulations and visualization of data.

**Comment by jinca** [comment]
Thanks again @ykvishal, could you please let me know the configuration of the modules you used to do the visualization? Cheers!

**Reply by ykvishal** [reply]
This part is tricky, and it is not related to MOOSE. You may have to work with your HPC support team. I had shared my following requirements to my HPC support team:
1. Use of desktop OnDemand (DOD)
2. Paraview is installed in DOD
3. DOD works on whole node (ie I can use all cores of a given node)
4. DOD also has a GPU
5. Paraview works in auto parallel mode in DOD (not using master/slave option. MS option is very slow)

After a lot of problems, my HPC support team managed to make it work for me. Now, I can easily process 4TB data.  Also, for a couple of GB data, normal paraview works fine. 

I hope that it will work out for you. All the best.
